packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -0,0 +1,153 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import classNames from 'classnames';
+import { Localize, localize } from '@deriv/translations';
+import { connect } from 'Stores/connect';
+import { CSSTransition } from 'react-transition-group';
+import { Icon, InputField, Text, ThemedScrollbars } from '@deriv/components';
+import Fieldset from 'App/Components/Form/fieldset.jsx';
+import BarriersList from './barriers-list.jsx';
+
+const BarrierSelector = ({
+    barrier_1,
+    current_focus,
+    onChange,
+    setHoveredBarrier,
+    setCurrentFocus,
+    turbos_barrier_choices,
+    validation_errors,
+}) => {
+    const [selected_barrier, setSelectedBarrier] = React.useState(barrier_1);
+    const [is_barriers_table_expanded, setIsBarriersTableExpanded] = React.useState(false);
+
+    const header_tooltip_text = (
+        <React.Fragment>
+            <p>
+                <span style={{ fontWeight: 'bold' }}>For Long: </span>You will earn profit if the market stays above the
+                entry spot and doesn&apos;t cross the barrier.
+            </p>
+            <br />
+            <p>
+                <span style={{ fontWeight: 'bold' }}>For Short: </span>You will earn profit if the market stays below
+                the entry spot and doesn&apos;t cross the barrier.
+            </p>
+        </React.Fragment>
+    );
+
+    const toggleBarriersTable = () => setIsBarriersTableExpanded(!is_barriers_table_expanded);
+
+    const changeBarrier = e => {
```

Comment:
@kate-deriv let's try to move this logics inside BarriersList so that we pass only the data we need (hovered barrier and clicked barrier) to this parent component😊